### PR TITLE
feat(grafana): report local barrier manager progress

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Add a new panel for "Earliest In-Flight Barrier Progress". Specifically, this is the number of actors that have processed the earliest in-flight barriers (then reported to the local barrier manager) per second. This metric helps users to detect potential congestion or stuck in the system.

image

If there's no new checkpoint done but the local progress is still non-zero, then the system is just slow but not stuck.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/stream/src/task/barrier_manager/managed_state.rs [hunk 1]
```diff
@@ -84,85 +85,71 @@ impl ManagedBarrierState {
...
-                            ChainState::ConsumingUpstream(consumed_epoch, _) => consumed_epoch,
-                            ChainState::Done => epoch,
-                        },
-                        consumed_rows: match state {
-                            ChainState::ConsumingUpstream(_, consumed_rows) => consumed_rows,
-                            ChainState::Done => 0,
-                        },
-                    })
-                    .collect();
+        while let Some(entry) = self.epoch_barrier_state_map.first_entry() {
```
Comment (BugenZhao) [inline src/stream/src/task/barrier_manager/managed_state.rs:93]: Few changes except for whitespaces.

**Review by hzxa21 — APPROVED** [review]
LGTM

If this metric is meant to be exposed to the user in additional to the developer, shall we expose it in the user dashboard as well?

**Comment by BugenZhao** [comment]
I guess only adding it to the developer dashboard is enough for now. 👀

File: grafana/risingwave-dev-dashboard.dashboard.py [hunk 2]
```diff
@@ -839,6 +839,17 @@ def section_streaming(panels):
                 ),
             ],
         ),
+        panels.timeseries_ops(
```
[the remark this answers is not in the document]
Comment (BugenZhao) [reply grafana/risingwave-dev-dashboard.dashboard.py:842]: I didn't find a better unit for this metric. 😂 I guess if we treat the update in the local barrier manager as an operation, then it seems to make sense.

**Pull request merged**